src/pypilecore/results/typing.py

```diff
@@ -0,0 +1,40 @@
+from typing import List, Protocol, TypeVar, runtime_checkable
+
+from pypilecore.common.piles import PileProperties
+from pypilecore.results.compression.multi_cpt_results import (
+    CPTCompressionGroupResultsTable,
+    SingleCPTCompressionBearingResultsContainer,
+)
+from pypilecore.results.tension.multi_cpt_results import (
+    CPTTensionGroupResultsTable,
+    SingleCPTTensionBearingResultsContainer,
+)
+
+T = TypeVar(
+    "T",
+    SingleCPTCompressionBearingResultsContainer,
+    SingleCPTTensionBearingResultsContainer,
+)
+S = TypeVar("S", CPTCompressionGroupResultsTable, CPTTensionGroupResultsTable)
```

Comment:
These classes also have similar methods, so you could also make a Protocol for them